### DISCUSSION TITLE
MyTourbook produce a speed peak after correcting the elevation

### DISCUSSION THREAD

**Discussion by ovahead** [mytourbook/mytourbook#1106] [General]
After using the elevation correction (start and end the same) I get a peak of 77,1 km/h in the speed.
I never saw this before so I think it must have to do with this specific tour...
I can't imagine that this is intentional ;)

Before:
![Before-correcting-the-height](https://github.com/mytourbook/mytourbook/assets/112829463/a1d90d38-82a5-47d1-8309-aabbba188b2d)

After:
![after-correcting](https://github.com/mytourbook/mytourbook/assets/112829463/31c52ca7-fc86-441a-9e6f-f7bcb404e982)

**Comment by ovahead** [comment]
A better graphic for the prove of speed of 77,1 km/h ;)
![Better-Graphic](https://github.com/mytourbook/mytourbook/assets/112829463/cb9cffba-1ebf-4483-bd31-e317d38624a2)

**Comment by wolfgang-ch** [comment]
When you would show time on the x-axis and not distance, there would be a pause at ~5.3 km where the speed algorithm can have problems when it is not smoothed enough. I welcome any improvements for the speed algorithm but it is not simple.

**Comment by ovahead** [comment]
Thanks. Yes, there was a 20 min pause. Okay, I will have a look at it with Fit File Repair Tool and maybe I delete same data at this point with a speed of 0 km (and same distance) and add a start/stop event. Even I already break the tour with my Garmin at this point... Maybe it helps 
Otherwise I omit the elevation correction for this tour

**Reply by ovahead** [reply]
FFRT cant change anything. 
It had also nothing to with the elevation correction. After saveing the file a second time (after naming it and adding weather and so on) MTB add this peak. No peak after saving it the first time...

**Reply by ovahead** [reply]
Is there a way to change the peak manually?

**Reply by wolfgang-ch** [reply]
> Is there a way to change the peak manually?

No, the speed value is computed from time and distance.

You can play with the smoothing parameters to improve your speed graph

![image](https://github.com/mytourbook/mytourbook/assets/1283445/5613fc77-01a7-498a-bae8-29159df26815)

![image](https://github.com/mytourbook/mytourbook/assets/1283445/2acde015-5bda-482b-aea9-a682dc16431b)

![image](https://github.com/mytourbook/mytourbook/assets/1283445/1fafbfa7-defe-41ce-a1be-d90c7eda1af9)

**Comment by ovahead** [comment]
Okay thanks!
Maybe it's time to revise the speed algorithm ;) I loaded the tour also in MyTourbook and in Turtle Sport and the there no such massive peaks. 
Anyway, unfortunately smoothing does not change the maximum speed in the tour info :( And as far as I know, when smoothing is activated, this also applies to all tours, doesn't it?

**Comment by ovahead** [comment]
Please, please change something about the speed calculation.
In one tour, MTB calculated a max speed of 174,71 km/h. With a mountain bike on a ground level. In another tour MTB calculated several peaks (6) between 44 and 54 km/h within one kilometer. There wasn't a longer break, but rather a relatively constant ride at around 20-24 km/h These peaks, with utopian values, are simply absolutely unacceptable.

**Comment by wolfgang-ch** [comment]
When you provide me an algorithm how to compute it correctly, then I'll implement it, currently you can select from 3 smoothing algorithms,

I'm not a math genius.

**Comment by ovahead** [comment]
Yes, sorry!

I activated the smoothing. Looks better. Although I'm not sure about the values. Do you have some hints for the values of the originally smoothing algorithm? What are good average values? Should I also enable the value limiting?

**Comment by wolfgang-ch** [comment]
Smoothing is documented in https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/tools/tour-chart-smoothing.

Suprisingly I received sometime this [PDF](https://mytourbook.sourceforge.io/mytourbook/files/mytourbook_smooth_11.8.2.pdf) from Didier Jamet where he described his algorithm, then I integrated it into MT
